src/contracts/FraxlendPairCore.sol

```diff
@@ -1268,4 +1311,80 @@ abstract contract FraxlendPairCore is FraxlendPairAccessControl, FraxlendPairCon
             externalAssetVault.whitelistUpdate(false);
         }
     }
+
+    /// @notice The ```_unwindAndSendCollateral``` function will unwind collateral asset, which for Peapods LVF pods is an
+    /// autocompounding LP ERC4626 asset. This is most significant and important for self lending pods in which the collateral
+    /// is made up of this pair's supply receipt, and this can be used to redeem assets before realizing bad debt on a liquidation.
+    /// @param _amount The number of collateral assets to unwind
+    /// @param _receiver The receiver of the unwound assets
+    /// @return _t0 The 0th asset in the liquidity pair
+    /// @return _t1 The 1st asset in the liquidity pair
+    /// @return _a0 The amount of the 0th asset we're returning
+    /// @return _a1 The amount of the 1st asset we're returning
+    function _unwindAndSendCollateral(uint256 _amount, address _receiver)
+        internal
+        returns (address _t0, address _t1, uint256 _a0, uint256 _a1)
+    {
+        // Check if withdraw is paused and revert if necessary
+        if (isWithdrawPaused) revert WithdrawPaused();
+
+        IERC4626(address(collateralContract)).redeem(_amount, address(this), address(this));
+        address _spTkn = IERC4626(address(collateralContract)).asset();
+        IStakingPoolToken(_spTkn).unstake(IERC20(_spTkn).balanceOf(address(this)));
+        address _uniV2Tkn = IStakingPoolToken(_spTkn).stakingToken();
+        _t0 = IUniswapV2Pair(_uniV2Tkn).token0();
+        _t1 = IUniswapV2Pair(_uniV2Tkn).token1();
+
+        address _pod = IStakingPoolToken(_spTkn).INDEX_FUND();
+        uint256 _uniV2Amt = IERC20(_uniV2Tkn).balanceOf(address(this));
+
+        uint256 _t0Before = IERC20(_t0).balanceOf(address(this));
+        uint256 _t1Before = IERC20(_t1).balanceOf(address(this));
+
+        IERC20(_uniV2Tkn).approve(_pod, _uniV2Amt);
+        IDecentralizedIndex(_pod).removeLiquidityV2(_uniV2Amt, 0, 0, block.timestamp);
+
+        _a0 = IERC20(_t0).balanceOf(address(this)) - _t0Before;
+        _a1 = IERC20(_t1).balanceOf(address(this)) - _t1Before;
+
+        // Calculate the number of assets to transfer based on the shares to burn
+        uint256 _shares = _t0 == address(this) ? _a0 : _a1;
```

Comment:
we need to make sure that at least `_t0` or `_t1` is `address(this)`. Otherwise, although I am not sure, for pods that doesn't use the frax pair as a lp component can steal. 

`require(_t0 == address(this) || _t1 == address(this));`